trieste/models/model_interfaces.py

```diff
@@ -379,6 +399,19 @@ def covariance_between_points(
 
         return cov
 
+    def get_observation_noise(self):
+        """
+        Return the variance of observation noise for homoscedastic likelihoods.
+        :return: The observation noise.
+        :raise NotImplementedError: If the model does not have a homoscedastic likelihood.
+        """
+        try:
+            noise_variance = self.model.likelihood.variance
+        except NotImplementedError:
```

Comment:
Are you sure this is right? Do GPModels always have a likelihood.variance? If not, does trying to access it result in a NotImplementedError or an AttributeError?

Reply:
You are right, have made changes